embassy-stm32/src/eth/mod.rs

```diff
@@ -39,6 +40,11 @@ impl<const TX: usize, const RX: usize> PacketQueue<TX, RX> {
             rx_buf: [Packet([0; RX_BUFFER_SIZE]); RX],
         }
     }
+
+    // Allow to initialize a Self without requiring it to go on the stack
+    pub unsafe fn init(this: &mut MaybeUninit<Self>) {
```

Comment:
I think this doesn't need to be unsafe, there's no safety invariant the user has to satisfy?

Reply:
Done in last commit